### DISCUSSION TITLE
How to handle critical code

### DISCUSSION THREAD

**Discussion by danies8** [quartznet/quartznet#2320] [Q&A]
Hi
I have critical code in job scheduler?
Do I need to protect him by mutex or samphore slim?
I'm using net 8,last version of quartz net?
Thanks,

**Comment by danies8** [comment]
Does using mutex, samphore slim...can cause deadlocks ?
and cause the scheduler to frozen.

**Reply by danies8** [reply]
@lahma  
Can you please answer?
It is very important.

**Reply by lahma** [reply]
You can deadlock a process with incorrect usage of locking primitives. It's your responsibility to use them right. Quartz doesn't know what to lock when you have your "critical code".

**Reply by danies8** [reply]
So using mutex or samphore slim in "criticl code" within job scheduler is possible?

**Reply by lahma** [reply]
Maybe you can try it out, there's also documentation available if you want to understand how the scheduler works.

**Reply by danies8** [reply]
1.I used logger to monitor the scheduler.
2.Now I removed samphore slim and test if the scheduler is frozen.
I have "criticl code" in the begining of job  scheduler + fire Task (system.Threading.Task) that do task as part of job scheduler and all Tasks also reach same critical code
3. I will try and update the ticket.
4.I want to know more how scheduler works, where I can find it ?

**Reply by danies8** [reply]
I have GUI in MAUI that insert all the jobs into my DB.
In worker service I read these jobs and fire the scheduler.
A big project.

**Reply by danies8** [reply]
@lahma 
Can you please answer ?

**Reply by lahma** [reply]
I'm very close to blocking you from here as I find your constant pinging abusive.

**Reply by danies8** [reply]
OK-:)
